batch_tasks/resource/hook/batch_tasks.py

```diff
@@ -0,0 +1,385 @@
+import logging
+
+import ftrack_api
+
+
+class BatchTasksAction(object):
+    '''Batch Tasks action
+
+    `label` a descriptive string identifing your action.
+
+    `varaint` To group actions together, give them the same
+    label and specify a unique variant per action.
+
+    `identifier` a unique identifier for your action.
+
+    `description` a verbose descriptive text for you action
+     '''
+    label = "Batch Tasks"
+    variant = None
+    identifier = "batch-tasks"
+    description = None
+
+    def __init__(self, session):
+        '''Expects a ftrack_api.Session instance'''
+
+        self.logger = logging.getLogger(
+            '{0}.{1}'.format(__name__, self.__class__.__name__)
+        )
+
+        if self.label is None:
+            raise ValueError(
+                'Action missing label.'
+            )
+
+        elif self.identifier is None:
+            raise ValueError(
+                'Action missing identifier.'
+            )
+
+        self._session = session
+
+    @classmethod
+    def clone_session(cls, session):
+        assert (
+            isinstance(session, ftrack_api.Session)
+        ), 'Must be ftrack_api.Session instance.'
+
+        return ftrack_api.Session(
+            session.server_url, session.api_key, session.api_user
+        )
+
+    def register(self):
+        '''Registers the action, subscribing the the discover and launch
+        topics.'''
+        self._session.event_hub.subscribe(
+            'topic=ftrack.action.discover', self._discover
+        )
+
+        self._session.event_hub.subscribe(
+            'topic=ftrack.action.launch and data.actionIdentifier={0}'.format(
+                self.identifier
+            ),
+            self._launch
+        )
+
+    def _discover(self, event):
+        args = self._translate_event(
+            self._session, event
+        )
+
+        accepts = self.discover(
+            self._session, *args
+        )
+
+        if accepts:
+            return {
+                'items': [{
+                    'label': self.label,
+                    'varian': self.variant,
+                    'description': self.description,
+                    'actionIdentifier': self.identifier,
+
+                }]
+            }
+
+    def discover(self, session, uid, entities, source, values, event):
```

Comment:
This was the recommended structure for the base action for the new api. Since I last checked the base action been updated, so these function argument can be reduced; https://bitbucket.org/ftrack/ftrack-action-handler/src/cdcf1994b672ca8a133a87cb40a458e9bf5d83da/source/ftrack_action_handler/action.py?at=master&fileviewer=file-view-default

Can't remove all as they are not keyword arguments.